File: JavaScript/7js 1/automorphic.js
/*
|--------------------------------------------------------------------------
| Automorphic Number (Special Numbers Series #6)
|--------------------------------------------------------------------------
|
*/

function automorphic(n) {
   return n == String(n ** 2).slice(-String(n).length) ? "Automorphic" : "Not!!";
}


function automorphic(n) {
   return `${n ** 2}`.endsWith(n) ? `Automorphic` : `Not!!`;
}


function automorphic(n) {
   return RegExp(`${n}$`).test(n ** 2) ? "Automorphic" : "Not!!";
}


function automorphic(n) {
   return Math.pow(n, 2).toString().endsWith(n.toString()) ? "Automorphic" : "Not!!";
}


console.log(automorphic(6), "Automorphic");
console.log(automorphic(9), "Not!!");
console.log(automorphic(76), "Automorphic");
console.log(automorphic(95), "Not!!");
console.log(automorphic(625), "Automorphic");
console.log(automorphic(225), "Not!!");

File: Python/7py 1/automorphic.py
"""
|--------------------------------------------------------------------------
| Automorphic Number (Special Numbers Series #6)
|--------------------------------------------------------------------------
|
"""

def automorphic(n):
   return "Automorphic" if str(n) == str(n**2)[-len(str(n)):] else "Not!!"


def automorphic(n):
   return ('Not!!', 'Automorphic')[str(n ** 2).endswith(str(n))]


def automorphic(n):
   return 'Automorphic' if (n ** 2) % 10 ** len(str(n)) == n else 'Not!!'


def automorphic(n):
   return "Automorphic" if str(n**2).endswith(str(n)) else "Not!!"


print(automorphic(6), "Automorphic")
print(automorphic(9), "Not!!")
print(automorphic(76), "Automorphic")
print(automorphic(95), "Not!!")
print(automorphic(625), "Automorphic")
print(automorphic(225), "Not!!")